File: 02SortingTechniques/SelectionSort/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
    static void selectionSort(int arr[]) {
        for (int i = 0; i < arr.length - 1; i++) {
            int s = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[s]) {
                    s = j;
                }
            }
            if (s != i) { // check if the current element is the smallest
                int t = arr[i];
                arr[i] = arr[s];
                arr[s] = t;
            }
        }
    }

    public static void main(String[] args) {
        int arr[] = { 1, 4, 5, 2, 5, 6, 5, 0, 2, 1 };
        System.out.println("Before : " + Arrays.toString(arr));
        selectionSort(arr);
        System.out.println("After : " + Arrays.toString(arr));
    }
}

File: 02SortingTechniques/SelectionSort/SelectionSort.cpp
#include <bits/stdc++.h>
using namespace std;
void selectionSort(vector<int> &arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        int s = i;
        for (int j = i + 1; j < arr.size(); j++)
        {
            if (arr[j] < arr[s])
            {
                s = j;
            }
        }
        if (s != i) // check if the current element is the smallest
        {
            // swap(arr[i], arr[s]);
            int t = arr[i];
            arr[i] = arr[s];
            arr[s] = t;
        }
    }
}
int main(void)
{
    vector<int> arr = {1, 4, 5, 2, 3, 7, 8, 4, 0};
    for (int ele : arr)
    {
        cout << ele << " ";
    }
    cout << endl;
    selectionSort(arr);
    for (int ele : arr)
    {
        cout << ele << " ";
    }
    cout << endl;
    return 0;
}